File: demo/src/main/java/com/example/demo/entity/Notificacion.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "tbl_notificaciones")
@Data
@AllArgsConstructor @NoArgsConstructor @Builder
public class Notificacion {

	public enum Estado {
		PENDIENTE, ENVIADO, ABORTADA, ERROR
	}

	public enum Medio {
		SMS, EMAIL
	}

	public enum Tipo {
		ANUNCIO_NOTA_ESTUDIANTE, ANUNCIO_AULA_ESTUDIANTE, ANUNCIO_AULA_VIGILANTE, PASSWORD_RESET
	}
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String asunto;

	private String cuerpo;

	private String emailDestino;

	@Temporal(TemporalType.TIMESTAMP)
	private Date programacionEnvio;

	@Enumerated(EnumType.STRING)
	private Medio[] medios;

	@Enumerated(EnumType.STRING)
	private Tipo tipo;

	@Enumerated(EnumType.STRING)
	private Estado estado;

	private String mensajeError;

	@Temporal(TemporalType.TIMESTAMP)
	private Date momentoRealEnvio;

}

File: Angular EnvioNotificaciones/practica5/src/app/models/notificacion.ts
export enum Medio {
    SMS = 'SMS',
    EMAIL = 'EMAIL'
  }
  
  export enum Tipo {
    ANUNCIO_NOTA_ESTUDIANTE = 'ANUNCIO_NOTA_ESTUDIANTE',
    ANUNCIO_AULA_ESTUDIANTE = 'ANUNCIO_AULA_ESTUDIANTE',
    ANUNCIO_AULA_VIGILANTE = 'ANUNCIO_AULA_VIGILANTE',
    PASSWORD_RESET = 'PASSWORD_RESET'
  }
  
  export enum Estado {
    PENDIENTE = 'PENDIENTE',
    ENVIADO = 'ENVIADO',
    ABORTADA = 'ABORTADA',
    ERROR = 'ERROR'
}

export class Notificacion {
    id!: number;
    asunto: string | undefined;
    cuerpo: string | undefined;
    emailDestino: string | undefined;
    telefonoDestino: string | undefined;
    programacionEnvio!: string;
    medios: Medio[] | undefined;
    tipo: Tipo | undefined;
    estado: Estado | undefined;
    mensajeError: string | undefined;
    momentoRealEnvio: string | undefined;
}